### PR TITLE
Add TerminatePhase action

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

We'll need this action to terminate the initialisation phase once it is composed of individual actions (see #1575).

I also add a commit in this PR that uses it for the `RegisterWithObservers` phase so that more registering actions can be done in that phase. An example is registering with the upcoming numeric data importer.

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Parallel/Actions/TerminatePhase.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
```
Comment (nilsdeppe) [inline src/Parallel/Actions/TerminatePhase.hpp:1]: There's also a `GotoAction.hpp` that's just in `src/Parallel/`. I'm fine with either having the `Actions` subdir or having the actions just in `src/Parallel` (though the `Actions` is more consistent). Would you mind moving the `Goto` header into the `Actions` directory?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
One small request

**Review by kidder — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]
Thanks for moving the `GoTo` action! :)

**Comment by kidder** [comment]
@nilsleiffischer one of the tests is failing all builds...

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**